internal/provider/physical_replication_stream.go

```diff
@@ -0,0 +1,384 @@
+		state.ActivatedAt = types.StringValue(apiObj.GetActivatedAt().String())
+	} else {
+		state.ActivatedAt = types.StringNull()
+	}
+	state.Status = types.StringValue(string(apiObj.GetStatus()))
+
+	if state.FailoverImmediately.ValueBool() {
+		state.FailoverImmediately = types.BoolValue(true)
+	}
+}
+
+func (r *physicalReplicationStreamResource) Read(
+	ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse,
+) {
+	if r.provider == nil || !r.provider.configured {
+		addConfigureProviderErr(&resp.Diagnostics)
+		return
+	}
+
+	var state PhysicalReplicationStream
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	id := state.ID.ValueString()
+
+	traceAPICall("GetPhysicalReplicationStream")
+	apiResp, httpResp, err := r.provider.service.GetPhysicalReplicationStream(ctx, id)
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == http.StatusNotFound {
+			resp.Diagnostics.AddWarning(
+				"PCR stream not found",
+				fmt.Sprintf(
+					"PCR stream with id %s is not found. Removing from state.",
+					id,
+				),
+			)
+			resp.State.RemoveResource(ctx)
+		} else {
+			resp.Diagnostics.AddError(
+				"Error getting PCR stream",
+				fmt.Sprintf("Could not get PCR stream: %s", formatAPIErrorMessage(err)),
+			)
+		}
+		return
+	}
+
+	loadPCRStreamToTerraformState(apiResp, &state)
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *physicalReplicationStreamResource) Update(
+	ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse,
+) {
+	if r.provider == nil || !r.provider.configured {
+		addConfigureProviderErr(&resp.Diagnostics)
+		return
+	}
+
+	var plan, state PhysicalReplicationStream
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Validate that exactly one of failover_at or failover_immediately is set
+	if plan.FailoverAt.IsNull() && plan.FailoverImmediately.IsNull() {
+		resp.Diagnostics.AddError(
+			"Invalid failover configuration",
+			"Exactly one of failover_at or failover_immediately must be set",
+		)
+		return
+	}
+	if !plan.FailoverImmediately.IsNull() && !plan.FailoverImmediately.ValueBool() {
+		resp.Diagnostics.AddError(
+			"Invalid failover configuration",
+			"If failover_immediately is set, it must be set to true",
+		)
+		return
+	}
+
+	// Create update spec with FAILING_OVER status
+	failingOverStatus := client.REPLICATIONSTREAMSTATUSTYPE_FAILING_OVER
+	updateSpec := &client.UpdatePhysicalReplicationStreamSpec{
+		Status: &failingOverStatus,
+	}
+
+	// If failover_at is specified in the plan, set it in the update spec
+	// Only set failover_at if failover_immediately is not true
+	if !plan.FailoverAt.IsNull() && !plan.FailoverImmediately.ValueBool() {
```

Comment:
nit: it might help readability if you pulled these out into some local variables.  Each of them has been used multiple times now.  Could be something like:
```
isFailoverAt := !plan.FailoverAt.IsNull()
isFailoverImmediately := !plan.FailoverImmediately.IsNull()

if isFailoverAt && isFailoverImmediately  {
 // invalid configuration
}

if isFailoverImmediately && !plan.FailoverImmediately.ValueBool() {
   // .. invalid value for failover immediately
}

// now you know that isFailoverImmediately also indicates the value is valid
if isFailoverAt {
   // set the failoverAt value on the spec
}
```